primary/src/block_synchronizer/mock.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
```

Comment:
It might be considered a bit far fetched building this mock, but I want to avoid writing for every test code to wait on channels and handling sending back results. Will have to test more cases on follow up PRs for the batch synchronization as well and this mock will also be re-used.

In general I would look into building a more generic mock for testing async components that communicate via channels